libraries/AP_NavEKF2/AP_NavEKF2_core.cpp

```diff
@@ -606,6 +608,18 @@ void NavEKF2_core::calcOutputStates()
     // apply a trapezoidal integration to velocities to calculate position
     outputDataNew.position += (outputDataNew.velocity + lastVelocity) * (imuDataNew.delVelDT*0.5f);
 
+    // calculate the average angular rate across the last IMU update
+    // note delAngDT is prevented from being zero in readIMUData()
+    Vector3f angRate = imuDataNew.delAng * (1.0f/imuDataNew.delAngDT);
```

Comment:
This has potential to create quite a lot of velocity noise. Should it not use filtered angular velocity?